torch/_dynamo/convert_frame.py

```diff
@@ -215,9 +215,26 @@ def _fn(*args: _P.args, **kwargs: _P.kwargs) -> _T:
             prior_warn_only = torch.is_deterministic_algorithms_warn_only_enabled()
             py_rng_state = random.getstate()
             torch_rng_state = torch.random.get_rng_state()
-            cuda_rng_state = None
+            cuda_rng_state: Optional[Union[torch.Tensor, torch._C.Generator]] = None
             if torch.cuda.is_available():
-                cuda_rng_state = torch.cuda.get_rng_state()
+                if not torch.cuda.is_current_stream_capturing():
+                    cuda_rng_state = torch.cuda.get_rng_state()
+                else:
+                    # torch.compile can be called under stream capture
+                    # for higher order operators when doing stream
+                    # capture to a graph in eager mode. For an
+                    # example, search "torch.compile" in
+                    # torch/_higher_order_ops.cond.py. This is
+                    # normally to verify specific attributes of the
+                    # called functions (e.g., that torch.cond's
+                    # true_fn and false_fn both return a tensor with
+                    # the same metadata), so it is
+                    # unavoidable. Presumably even under stream
+                    # capture, we still want to save and restore the
+                    # rng state.
+                    cuda_rng_state = torch.cuda.default_generators[
```

Comment:
TODO: Make sure that resetting the cuda rng state works as intended when doing stream capture. Frankly, I'm not even sure what the intended behavior is supposed to be. This is the first time that a code sequence like:

```
with torch.cuda.graph(g):
    torch.compile(f)
```

is being done.